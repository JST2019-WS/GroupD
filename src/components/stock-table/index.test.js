/* eslint-env node, jest */

import StockTable from "./index";
import {shallow, mount} from "enzyme";
import stocks from '../../../test/stocks.json'

describe('Stock Table', () => {
    it('should render as many stocks as passed', () => {
        const table = shallow(<StockTable stocks={['a', 'b', 'c']}/>);
        expect(table.find('StockTableRow')).toHaveLength(3);
        expect(table.debug()).toMatchSnapshot();
    });

    it('should handle empty/null stock arrays', () => {
        let table = shallow(<StockTable stocks={null}/>);
        expect(table.html()).toMatchSnapshot('null-stocks');

        table = shallow(<StockTable stocks={[]}/>);
        expect(table.html()).toMatchSnapshot('null-stocks');
    });

    it('should propagate stock click/hover events', () => {
        const onClick = jest.fn();
        const onHover = jest.fn();
        const table = mount(<StockTable stocks={stocks} onStockClicked={onClick} onStockHovered={onHover} />);

        stocks.forEach((elem, id) => {
            table.find('StockTableRow').at(id).prop('onClick')();
            expect(onClick.mock.calls).toHaveLength(id+1);
            expect(onClick.mock.calls[id][0]).toBe(elem);
        });

        stocks.forEach((elem, id) => {
            table.find('StockTableRow').at(id).prop('onHover')();
            expect(onHover.mock.calls).toHaveLength(id+1);
            expect(onHover.mock.calls[id][0]).toBe(elem);
        });
    });

    it('should handle null click/hover callbacks', () => {
        const table = mount(<StockTable stocks={stocks} onStockClicked={null} onStockHovered={null} />);
        expect(table.html()).toMatchSnapshot('valid-stocks');

        // Click / Hover should not throw
        stocks.forEach((elem, id) => {
            table.find('StockTableRow').at(id).prop('onClick')();
            table.find('StockTableRow').at(id).prop('onHover')();
        });
    })
});