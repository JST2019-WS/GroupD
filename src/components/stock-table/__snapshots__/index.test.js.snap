// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stock Table should render as many stocks as passed 1`] = `
"<table className=\\"stock-table\\">
  <thead className=\\"stock-table-head\\">
    <tr>
      <th>
        Wertpapier
      </th>
      <th>
        Branche
      </th>
      <th>
        (Letzter) Kurs
      </th>
      <th>
        Abs.
      </th>
      <th>
        Perf. %
      </th>
      <th>
        Letztes Update
      </th>
      <th>
        Börse
      </th>
      <th>
        Umsatz
      </th>
    </tr>
  </thead>
  <tbody className=\\"stock-table-body\\">
    <StockTableRow stock=\\"a\\" onClick={[Function: bound ]} onHover={[Function]} />
    <StockTableRow stock=\\"b\\" onClick={[Function: bound ]} onHover={[Function]} />
    <StockTableRow stock=\\"c\\" onClick={[Function: bound ]} onHover={[Function]} />
  </tbody>
</table>"
`;

exports[`null-stocks 1`] = `"<table class=\\"stock-table\\"><thead class=\\"stock-table-head\\"><tr><th>Wertpapier</th><th>Branche</th><th>(Letzter) Kurs</th><th>Abs.</th><th>Perf. %</th><th>Letztes Update</th><th>Börse</th><th>Umsatz</th></tr></thead><tbody class=\\"stock-table-body\\"><tr>No stocks found.</tr></tbody></table>"`;

exports[`null-stocks 2`] = `"<table class=\\"stock-table\\"><thead class=\\"stock-table-head\\"><tr><th>Wertpapier</th><th>Branche</th><th>(Letzter) Kurs</th><th>Abs.</th><th>Perf. %</th><th>Letztes Update</th><th>Börse</th><th>Umsatz</th></tr></thead><tbody class=\\"stock-table-body\\"><tr>No stocks found.</tr></tbody></table>"`;

exports[`valid-stocks 1`] = `"<table class=\\"stock-table\\"><thead class=\\"stock-table-head\\"><tr><th>Wertpapier</th><th>Branche</th><th>(Letzter) Kurs</th><th>Abs.</th><th>Perf. %</th><th>Letztes Update</th><th>Börse</th><th>Umsatz</th></tr></thead><tbody class=\\"stock-table-body\\"><tr class=\\"stock-table-row\\"><td class=\\"name\\"><a href=\\"https://www.wallstreet-online.de/aktien/deutsche-pfandbriefbank-aktie\\">Deutsche Pfandbriefbank</a></td><td class=\\"category\\"><a href=\\"https://www.wallstreet-online.de/aktien/branche/banken-aktien\\">Banken</a></td><td class=\\"value\\">10,71 EUR</td><td class=\\"absolute negativeNumber\\">-0,26 EUR</td><td class=\\"relative negativeNumber\\">-2,37 %</td><td class=\\"timestamp small\\">04:57:06</td><td class=\\"exchange small\\">LAS</td><td class=\\"revenue small\\">35.729 EUR</td></tr><tr class=\\"stock-table-row\\"><td class=\\"name\\"><a href=\\"https://www.wallstreet-online.de/aktien/commerzbank-aktie\\">Commerzbank</a></td><td class=\\"category\\"><a href=\\"https://www.wallstreet-online.de/aktien/branche/banken-aktien\\">Banken</a></td><td class=\\"value\\">5,37 EUR</td><td class=\\"absolute positiveNumber\\">+0,14 EUR</td><td class=\\"relative positiveNumber\\">+2,67 %</td><td class=\\"timestamp small\\">04:57:06</td><td class=\\"exchange small\\">LAS</td><td class=\\"revenue small\\">549.033 EUR</td></tr><tr class=\\"stock-table-row\\"><td class=\\"name\\"><a href=\\"https://www.wallstreet-online.de/aktien/new-work-aktie\\">New Work</a></td><td class=\\"category\\"><a href=\\"https://www.wallstreet-online.de/aktien/branche/software-internet-aktien\\">Software und Internet</a></td><td class=\\"value\\">263,00 EUR</td><td class=\\"absolute negativeNumber\\">-9,00 EUR</td><td class=\\"relative negativeNumber\\">-2,94 %</td><td class=\\"timestamp small\\">04:57:06</td><td class=\\"exchange small\\">ETR</td><td class=\\"revenue small\\">527.432 EUR</td></tr><tr class=\\"stock-table-row\\"><td class=\\"name\\"><a href=\\"https://www.wallstreet-online.de/aktien/telefonica-deutschland-holding-aktie\\">Telefonica Deutschland Holding</a></td><td class=\\"category\\"><a href=\\"https://www.wallstreet-online.de/aktien/branche/telekommunikation-aktien\\">Telekommunikation</a></td><td class=\\"value\\">2.501,63 EUR</td><td class=\\"absolute negativeNumber\\">-0,33 EUR</td><td class=\\"relative negativeNumber\\">-0,20 %</td><td class=\\"timestamp small\\">04:57:06</td><td class=\\"exchange small\\">ETR</td><td class=\\"revenue small\\">9.917.848 EUR</td></tr></tbody></table>"`;
